### PR TITLE
added endpoint for retrieving details for the given Helm chart.

### PR DESCRIPTION
**Pull request by pedjak**
* url `/api/helm/chart`
   * `url` query param specifies the chart url
   * json encoded response

### CODE REVIEW

**Comment by pedjak** [comment]
This PR depends on https://github.com/openshift/console/pull/4088 and it needs to get merged first

**Comment by pedjak** [comment]
This PR depends on https://github.com/openshift/console/pull/4088 and it needs to get merged first

File: pkg/helm/actions/get_chart.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+)
+
+func GetChart(name string, conf *action.Configuration) (*chart.Chart, error) {
+
+	cmd := action.NewInstall(conf)
+
+	chartLocation, err := cmd.ChartPathOptions.LocateChart(name, settings)
```
[the remark this answers is not in the document]
Comment (jhadvig) [reply pkg/helm/actions/get_chart.go]: should we swap `name` with `url` here?

File: pkg/helm/handlers/handler_test.go [hunk 2]
```diff
@@ -49,6 +50,12 @@ func fakeGetManifest(mockedManifest string, err error) func(name string, url str
 	}
 }
 
+func mockedHelmGetChart(c *chart.Chart, e error) func(name string, conf *action.Configuration) (*chart.Chart, error) {
```
Comment (jhadvig) [inline pkg/helm/handlers/handler_test.go]: should we swap `name` with `url` here?

File: pkg/helm/handlers/handler_test.go [hunk 3]
```diff
@@ -49,6 +50,12 @@ func fakeGetManifest(mockedManifest string, err error) func(name string, url str
...
+	return func(name string, conf *action.Configuration) (*chart.Chart, error) {
```
Comment (jhadvig) [inline pkg/helm/handlers/handler_test.go]: same

File: pkg/helm/actions/get_chart_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package actions
+
+import (
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chartutil"
+	kubefake "helm.sh/helm/v3/pkg/kube/fake"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"io/ioutil"
+	"testing"
```
Comment (jhadvig) [inline pkg/helm/actions/get_chart_test.go]: lets format there imports as we do in other pkgs

File: pkg/helm/handlers/handlers.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package handlers
 import (
 	"encoding/json"
 	"fmt"
+	"helm.sh/helm/v3/pkg/chart"
```
Comment (jhadvig) [inline pkg/helm/handlers/handlers.go]: would separate this import as standalone

File: pkg/helm/handlers/handlers.go [hunk 6]
```diff
@@ -100,3 +103,20 @@ func (h *helmHandlers) HandleHelmList(user *auth.User, w http.ResponseWriter, r
 	res, _ := json.Marshal(resp)
 	w.Write(res)
 }
+
+func (h *helmHandlers) HandleChartGet(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	params := r.URL.Query()
+	chartUrl := params.Get("url")
+
+	conf := h.getActionConfigurations(h.ApiServerHost, "default", user.Token, &h.Transport)
```
Comment (jhadvig) [inline pkg/helm/handlers/handlers.go]: why it's always "default" ?

lets add a comment at least to clarify

**Comment by sbose78** [comment]
Associated jira https://issues.redhat.com/browse/APPSVC-387

File: pkg/helm/actions/get_chart.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedjak) [reply pkg/helm/actions/get_chart.go]: `settings` is a global var in https://github.com/pedjak/console/blob/c54d7e8a8e951a3e7e89bb8e4848527674908458/pkg/helm/actions/config.go#L14

**Comment by jhadvig** [comment]
@benjaminapetersen could you please take a look/approve this one. Thanks !

**Comment by rohitkrai03** [comment]
/retitle Bug 1808524: added endpoint for retrieving details for the given Helm chart

**Comment by pedjak** [comment]
@rohitkrai03 I have solved conflicts and created PR: https://github.com/openshift/console/pull/4563

**Pull request merged**